airflow/www/app.py

```diff
@@ -980,25 +984,27 @@ def action(self):
                     t.task_id
                     for t in task.get_flat_relatives(upstream=True)]
             TI = models.TaskInstance
+            dates = utils.date_range(start_date, end_date)
```

Comment:
What feels like a reasonable cap to you? Since we already ask for confirmation I'm tempted to omit a cap since it simplifies things. I agree the cap, if it exists, should be on the cartesian product ```tasks``` object.